platform-hub-api/app/services/announcements_processor_service.rb

```diff
@@ -15,16 +15,22 @@ def process announcement
     @logger.info "Processing announcement #{announcement.id} - #{announcement.level} - '#{announcement.title}'"
 
     begin
-      announcement.update! status: :delivering
 
-      unless announcement.deliver_to.blank?
-        process_for_email_delivery announcement
-        process_for_slack_delivery announcement
+      announcement.with_lock do
```

Comment:
Question for @marcinc and @Joseph-Irving – this now causes the whole deliver of the announcement to be in a transaction… so any failure will rollback the updates and make it so the announcement is picked up to be processed again. This means that announcements will now be retried regardless of how far they got in the previous (failed) attempt. Is this a good?

A part of me thinks this is useful for handling intermittent issues. Another part of me is concerned that if there's a bug, we'll keep retrying and partial sending of this announcement will occur each time (e.g. 1st -> it sends emails but fails on Slack, 2nd -> it sends email, again, but fails on Slack again… and so on). That's probably not a good thing?